### PR TITLE
Fix Ids collisions in tests

### PR DESCRIPTION
**Pull request by hsmett**
See https://circleci.com/gh/HybirdCorp/creme_crm/184?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Fixes https://github.com/HybirdCorp/creme_crm/issues/14

### CODE REVIEW

File: creme/creme_config/tests/transfer/test_importing.py [hunk 1]
```diff
@@ -1311,7 +1311,7 @@ def test_customfields02(self):
         response = self.assertPOST200(self.URL, data={'config': json_file})
         self.assertFormError(
             response, 'form', 'config',
-            _('This custom-field type is invalid: {}.').format(1024)
+            _('This custom-field type is invalid: {}.').format(self.UNUSED_PK)
         )
```
Comment (genglert) [inline creme/creme_config/tests/transfer/test_importing.py:1315]: Here the custom field type corresponds to a constants in the code, not a PK which could be reached if sequences at not reset. It works but the name 'PK' if not totally accurate (I have no better proposition sorry). Hummm

File: creme/creme_config/tests/transfer/test_importing.py [hunk 2]
```diff
@@ -2263,7 +2263,7 @@ def test_entityfilters_error01(self):
         "Invalid condition type."
         self.login(is_staff=True)
 
-        cond_type = 1024
+        cond_type = self.UNUSED_PK
```
Comment (genglert) [inline creme/creme_config/tests/transfer/test_importing.py:2266]: Same remark that above

File: creme/creme_core/tests/views/test_search.py [hunk 3]
```diff
@@ -162,7 +162,7 @@ def test_search04(self):
         self.assertEqual(_('Please enter at least {count} characters').format(count=3),
                          self._search('ox').context['error_message']
                         )
-        self.assertEqual(404, self._search('linus', 1024).status_code)  # ct_id=1024 DOES NOT EXIST
+        self.assertEqual(404, self._search('linus', self.UNUSED_PK).status_code)  # ct_id DOES NOT EXIST
```
Comment (genglert) [inline creme/creme_core/tests/views/test_search.py:165]: I presume the comment is now useless

File: creme/persons/tests/contact/test_main.py [hunk 4]
```diff
@@ -550,7 +550,7 @@ def test_create_linked_contact06(self):
         user = self.login()
         orga = Organisation.objects.create(user=user, name='Acme')
 
-        self.assertGET404(self._build_addrelated_url(1024,  # Doesn't exist
+        self.assertGET404(self._build_addrelated_url(self.UNUSED_PK,  # Doesn't exist
```
Comment (genglert) [inline creme/persons/tests/contact/test_main.py:553]: Useless comment

File: creme/reports/tests/core/test_report.py [hunk 5]
```diff
@@ -184,7 +184,7 @@ def test_custom_field02(self):
         self.assertIs(hand.hidden, True)
 
     def test_custom_field_error(self):
-        cf_id = '1024'
+        cf_id = f'{self.UNUSED_PK}'
```
Comment (genglert) [inline creme/reports/tests/core/test_report.py:187]: str(self.UNUSED_PK)

File: creme/creme_config/tests/transfer/test_importing.py (hunk 1 above)
Comment (hsmett) [reply creme/creme_config/tests/transfer/test_importing.py:1315]: I could use a test-specific value here and first check that it's not an available choice of the formfield ?

Comment (hsmett) [reply creme/creme_config/tests/transfer/test_importing.py:1315]: Doesn't seem trivial with the ImportForm

Comment (genglert) [reply creme/creme_config/tests/transfer/test_importing.py:1315]: yes of course (leave 1024 or add a named variable/attribute)

**Comment by hsmett** [comment]
updated

File: creme/creme_core/tests/utils/test_main.py [hunk 6]
```diff
@@ -104,7 +104,7 @@ def test_truncate_str_01(self):
 
     def test_create_if_needed(self):
         title = 'Mister'
-        pk = 1024
+        pk = 999999
```
Comment (genglert) [inline creme/creme_core/tests/utils/test_main.py:107]: Missing refactoring to UNUSED_PK ??

Comment (hsmett) [reply creme/creme_core/tests/utils/test_main.py:107]: It's a create_if_needed so we need an eligible (< 2147483647 int) PK here

File: creme/creme_core/tests/views/test_entity_filter.py [hunk 7]
```diff
@@ -1866,7 +1866,7 @@ def find_user(u):
         ).json()
         self.assertEqual(choices1, choices3)
 
-        self.assertGET404(url + '?filter_type=1024')
+        self.assertGET404(url + f'?filter_type={self.UNUSED_PK}')
```
Comment (genglert) [inline creme/creme_core/tests/views/test_entity_filter.py:1869]: Not a PK (see EF_CREDENTIALS/EF_USER with are currently 0/1)

File: creme/cti/tests.py [hunk 8]
```diff
@@ -114,7 +114,7 @@ def test_add_phonecall02(self):
         "No contact."
         self.login()
 
-        self.assertPOST404(self.ADD_PCALL_URL, data={'entity_id': '1024'})
+        self.assertPOST404(self.ADD_PCALL_URL, data={'entity_id': f"{self.UNUSED_PK}"})
```
Comment (genglert) [inline creme/cti/tests.py:117]: str(self.UNUSED_PK)

File: creme/emails/tests/test_mail.py [hunk 9]
```diff
@@ -912,7 +912,7 @@ def test_resend02(self):
 
         url = reverse('emails__resend_emails')
         self.assertPOST409(url, data={'ids': 'notanint'})
-        self.assertPOST404(url, data={'ids': '1024'})
+        self.assertPOST404(url, data={'ids': f'{self.UNUSED_PK}'})
```
Comment (genglert) [inline creme/emails/tests/test_mail.py:915]: str(self.UNUSED_PK)

File: creme/reports/tests/test_forms.py [hunk 10]
```diff
@@ -1158,7 +1158,7 @@ def test_clean_error02(self):
                     'cell_key': cell.key,
                 },
                 'graph_type': {
-                    'type_id': 1024,  # <==
+                    'type_id': self.UNUSED_PK,  # <==
```
Comment (genglert) [inline creme/reports/tests/test_forms.py:1161]: Not a PK

File: creme/creme_core/tests/base.py [hunk 11]
```diff
@@ -710,6 +709,8 @@ def http_file(file_path):
 
 
 class CremeTestCase(TestCase, _CremeTestCase):
+    UNUSED_PK = sys.maxsize
```
Comment (genglert) [inline creme/creme_core/tests/base.py:712]: In _CremeTestCase to get it in CremeTransactionTestCase too no ?

File: creme/creme_core/tests/utils/test_main.py (hunk 6 above)
Comment (genglert) [reply creme/creme_core/tests/utils/test_main.py:107]: Ok ; so a comment would be nice to avoid a silly change in the future

**Comment by hsmett** [comment]
updated

File: creme/creme_core/tests/views/test_entity_filter.py [hunk 12]
```diff
@@ -1866,7 +1866,7 @@ def find_user(u):
         ).json()
         self.assertEqual(choices1, choices3)
 
-        self.assertGET404(url + '?filter_type=1024')
+        self.assertGET404(url + f'?filter_type=1024')
```
Comment (genglert) [inline creme/creme_core/tests/views/test_entity_filter.py:1869]: remove "f" prefix (or add a variable invalid_filter_type = 1024)

Comment (hsmett) [reply creme/creme_core/tests/views/test_entity_filter.py]: done

**Pull request merged**